components/datetime/src/format/neo.rs

```diff
@@ -2002,24 +2002,41 @@ impl<R: DateTimeNamesMarker> RawDateTimeNames<R> {
             let field = match item {
                 FieldForDataLoading::Field(field) => field,
                 FieldForDataLoading::TimeZone(time_zone) => {
-                    self.load_time_zone_essentials(zone_essentials_provider, locale)?;
                     match time_zone {
                         // `z..zzz`
                         ResolvedNeoTimeZoneSkeleton::SpecificShort => {
+                            self.load_time_zone_essentials(zone_essentials_provider, locale)?;
```

Comment:
Suggestion: make a bool `needs_time_zone_essentials` and then have just one call site into `load_time_zone_essentials`.